Fix ank flickering with additional wrap and truncate

inf17101 opened the pull request:
Issues: #250 

<!--  Description of the change in case no issue is mentioned -->

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

inf17101 commented:
@windsource: This PR includes the additional wrap/truncate functionalityon on top of PR #275. So, it can be decided if the PR with or without the wrap/truncate shall be used or not.

inf17101 commented:
@krucod3: As discussed we do not adapt any further requirements for changing the table output format. The changed reqs inside this PR do only cover the reqs for changing the Execution State output when retries are being executed.

inf17101 commented on ank/src/cli_commands/workload_table_row.rs:
```diff
@@ -94,4 +118,29 @@ mod tests {
     }
+
+    #[test]
+    fn utest_additional_info_msg_without_new_lines() {
```
To which requirement shall I link this utest? To `// [utest->swdd~cli-shall-present-workloads-as-table~1]` and leaving a comment in the requirement about the stripped info message?

inf17101 replied:
Ok done

inf17101 commented on ank/src/cli_commands/workload_table.rs:
```diff
@@ -0,0 +1,315 @@
+
+    fn truncate_table_column(
+        mut table: Table,
+        column_position: usize,
+        remaining_terminal_width: usize,
```
Shall I introduce a new type to prevent accidentially swapped arguments ?

inf17101 replied:
Done

inf17101 commented on ank/src/cli_commands/workload_table.rs:
```diff
@@ -0,0 +1,315 @@
+
+    fn wrap_table_column(
+        mut table: Table,
+        column_position: usize,
+        remaining_terminal_width: usize,
```
Same here, shall I introduce a new type to prevent accidentially swapped arguments ?

inf17101 replied:
Done

krucod3 commented on ank/src/cli_commands/workload_table.rs:
```diff
@@ -0,0 +1,304 @@
+        let default_table = Self::default_table(&self.rows);
+
+        let total_table_width: usize = default_table.total_width();
+        let remaining_terminal_width =
```
We should maybe rename for better readability: `remaining_terminal_width` -> `available_additional_info_width`

inf17101 replied:
Done

krucod3 commented on ank/src/cli_commands/workload_table.rs:
```diff
@@ -0,0 +1,304 @@
+    fn truncate_table_column(
+        mut table: Table,
+        column_position: usize,
+        remaining_terminal_width: usize,
```
This is actually the new column width, and I would propose to also call it this way: `new_column_width`

inf17101 replied:
Done

krucod3 reviewed (commented):
Looks good, documented some small findings.

inf17101 commented on agent/doc/swdesign/README.md:
```diff
@@ -779,20 +779,20 @@ Needs:
 
 Comment:
-For details on the runtime connector specific actions, e.g., create, see the specific runtime connector workflows.
+For details on the runtime connector specific actions, e.g. create, see the specific runtime connector workflows. In case of a workload creation error, the execution state is set to `Pending(Starting)` with additional information indicating that the system will retry creation.
```
Done

inf17101 commented:
> Looks good, documented some small findings.

I have addressed all comments with further commits. Some comments are duplicated on my screen, so I have answered to those having a reply field available in the WebUI.

krucod3 reviewed (approved):
LGTM 👍
